### PR TITLE
Feature: Move file viewer to be under task file tab

### PR DESCRIPTION
**Pull request by mlunoe**
This PR moves the file viewer to be underneath the task file tab.

This is the first step in preparation for the new LogViewer tab (new API).

Before:
~~https://cl.ly/002X0k0G342O/Screen%20Recording%202016-09-27%20at%2017.24.gif~~

After:
~~https://cl.ly/393V3Q29201K/Screen%20Recording%202016-09-27%20at%2017.23.gif~~

What it looks like after rebase with `cnvs-upgrade/master`:

### CODE REVIEW

**Comment by rcorral** [comment]
@mlunoe I don't think you want to point this PR to master

**Comment by rcorral** [comment]
sorry I'm such a n00b at github

File: src/js/pages/task-details/TaskDetail.js [hunk 1]
```diff
@@ -249,6 +264,31 @@ class TaskDetail extends mixin(InternalStorageMixin, TabsMixin, StoreMixin) {
...
+    let innerPath = TaskDirectoryStore.get('innerPath').split('/');
+    let onClickPath = '';
+    let crumbs = innerPath.map((directoryItem, index) => {
+      let textValue = directoryItem;
+
+      // First breadcrumb is always 'Working Directory'.
+      if (index === 0) {
+        textValue = 'Working Directory';
+      } else {
+        onClickPath += ('/' + directoryItem);
```
Comment (rcorral) [inline src/js/pages/task-details/TaskDetail.js]: Could you do this instead?
```
`/${directoryItem}`
```

Comment (rcorral) [inline src/js/pages/task-details/TaskDetail.js]: I don't think the breadcrumbs are working as you intended
img

Comment (mlunoe) [inline src/js/pages/task-details/TaskDetail.js]: Fixed!

Comment (mlunoe) [inline src/js/pages/task-details/TaskDetail.js]: Done!

File: src/js/routes/services.js [hunk 2]
```diff
@@ -147,34 +146,45 @@ let serviceRoutes = {
...
+                  ]
                 },
+                // {
+                //   type: Route,
+                //   name: 'services-task-details-logs',
+                //   dontScroll: true,
+                //   path: 'logs/:filePath?/?:innerPath?/?',
+                //   handler: TaskLogsTab,
+                //   title:'Logs'
+                // },
```
Comment (mlunoe) [inline src/js/routes/services.js]: Done!

**Comment by mlunoe** [comment]
@rcorral addressed all comments

File: src/js/pages/task-details/TaskDetail.js [hunk 3]
```diff
@@ -261,12 +301,17 @@ class TaskDetail extends mixin(InternalStorageMixin, TabsMixin, StoreMixin) {
     }
 
     return (
-      <RouteHandler
-        directory={directory}
-        onOpenLogClick={this.handleOpenLogClick.bind(this)}
-        selectedLogFile={selectedLogFile}
-        service={this.getService()}
-        task={task} />
+      <div className="flex-container-col flex-grow flex-shrink">
```
Comment (jfurrow) [inline src/js/pages/task-details/TaskDetail.js]: These flex classes have changed a bit:
`flex-container-col` === `flex flex-direction-top-to-bottom` (two classes)
`flex-grow` === `flex-item-grow-1`
`flex-shrink` === `flex-item-shrink-1`
`flex-box` === `flex`

**Comment by jfurrow** [comment]
I think the code looks great. I did notice two bugs:
1. I navigated to the "files" tab for a task, then clicked the browser's "back" button. The "files" tab remained active, but the content was actually the "details" tab content.
2. The breadcrumb is visible in the "details" tab view.

**Comment by mlunoe** [comment]
Ok, both problems have been fixed:

**Comment by jfurrow** [comment]
Nice @mlunoe! I think maybe you need to update your local `cnvs-upgrade/master` because the conflict is still here :)

**Comment by mlunoe** [comment]
Nope, it was just squashed commits that was causing me trouble. Should be fixed now!

**Pull request merged**